corehq/apps/domain/views.py

```diff
@@ -1951,9 +1950,9 @@ def post(self, request, *args, **kwargs):
             elif request.POST.get('old_documentation_file', False):
                 new_domain.documentation_file_path = old.documentation_file_path
                 new_domain.documentation_file_type = old.documentation_file_type
-            new_domain.save()
 
             if publish_on_submit:
+                new_domain.save()
                 _publish_snapshot(request, self.domain_object, published_snapshot=new_domain)
```

Comment:
just to check, did you test this on domains that had images and documentation files? it's possible the extra saves had to do with how couch deals with attachments and doc update conflicts and would be good to make sure that they weren't added by design